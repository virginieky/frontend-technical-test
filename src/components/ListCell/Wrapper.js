import styled from 'styled-components';

const Wrapper = styled.li`
  width: 100%;
  height: 100px;
  flex: 0 0 auto;
  margin-bottom: 16px;
  margin-right: 16px;
`;

export default Wrapper;
