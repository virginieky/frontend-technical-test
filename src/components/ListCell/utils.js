export const getCellClick =
  ({ onClick, id }) =>
  () => {
    onClick(id);
  };
