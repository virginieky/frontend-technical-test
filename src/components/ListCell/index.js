import React from 'react';
import PropTypes from 'prop-types';

import { getDisplayedDate } from '../../utils';
import Padded from '../Padded';
import Text from '../Text';
import Wrapper from './Wrapper';
import { getCellClick } from './utils';

const ListCell = ({
  id,
  lastMessageTimestamp,
  onClick,
  recipientId,
  recipientNickname,
  senderId,
  senderNickname,
}) => {
  return (
    <Wrapper onClick={getCellClick({ onClick, id })}>
      <Padded right bottom left>
        <Text fontSize='16px' ellipsis>
          {id} {recipientId} {recipientNickname} - {senderId} {senderNickname}
        </Text>
        <Padded bottom small />
        <Text fontSize='12px' ellipsis>
          Last message sent: {getDisplayedDate(lastMessageTimestamp)}
        </Text>
      </Padded>
    </Wrapper>
  );
};

ListCell.defaultProps = {
  id: null,
  lastMessageTimestamp: null,
  onClick: () => {},
  recipientId: null,
  recipientNickname: null,
  senderId: null,
  senderNickname: null,
};

ListCell.propTypes = {
  id: PropTypes.number,
  lastMessageTimestamp: PropTypes.number,
  onClick: PropTypes.func,
  recipientId: PropTypes.number,
  recipientNickname: PropTypes.string,
  senderId: PropTypes.number,
  senderNickname: PropTypes.string,
};

export default ListCell;
