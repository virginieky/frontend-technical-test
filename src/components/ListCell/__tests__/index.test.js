import React from 'react';
import { render } from '@testing-library/react';
import ListCell from '../index';

describe('components | ListCell', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<ListCell />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should use onClick defaultProps', () => {
    const {
      defaultProps: { onClick },
    } = ListCell;

    expect(onClick).toBeDefined();
    expect(onClick({ preventDefault: jest.fn() })).toBe(undefined);
  });
});
