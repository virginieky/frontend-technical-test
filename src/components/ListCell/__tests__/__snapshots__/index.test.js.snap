// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCell should match the snapshot 1`] = `
<DocumentFragment>
  <li
    class="Wrapper-sc-m0e7wi-0 gFNerB"
  >
    <div
      class="Padded-sc-q6j7xl-0 chxjCd"
    >
      <p
        class="Text-sc-yw3xjc-0 gQhyGc"
        color="black"
        font-size="16px"
        font-weight="regular"
      >
           -  
      </p>
      <div
        class="Padded-sc-q6j7xl-0 dGTENR"
      />
      <p
        class="Text-sc-yw3xjc-0 fPFqtQ"
        color="black"
        font-size="12px"
        font-weight="regular"
      >
        Last message sent: 
      </p>
    </div>
  </li>
</DocumentFragment>
`;
